v0/destinations/facebook_pixel/network-handler.js

```diff
@@ -0,0 +1,142 @@
+const { isEmpty } = require("lodash");
+const {
+  processAxiosResponse,
+  getDynamicMeta
+} = require("../../../adapters/utils/networkUtils");
+const {
+  prepareProxyRequest,
+  proxyRequest
+} = require("../../../adapters/network");
+const { DESTINATION } = require("./config");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const ErrorBuilder = require("../../util/error");
+
+const defaultStatTags = {
+  destType: DESTINATION,
+  stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.RESPONSE_TRANSFORM,
+  scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+  meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.ABORTABLE
+};
+/**
+ * The actual API reference doc to which events from Rudderstack are being sent
+ * https://developers.facebook.com/docs/marketing-api/reference/ads-pixel/events/v13.0
+ *
+ * The documents referred while formulating the error responses
+ * 1. https://developers.facebook.com/docs/graph-api/guides/error-handling/
+ *   - This seems like a generic document that contains errors possible for Graph API
+ * 2. https://developers.facebook.com/docs/marketing-api/error-reference/
+ *   - The doc seems to be more related to Marketing API
+ */
+const RETRYABLE_ERROR_CODES = [1, 2, 3, 4, 341, 368, 5000, 190];
+
+/**
+ * Only under below mentioned scenario(s), add the errorCodes, subCodes etc,. to this map
+{
+  // A scenario where in we have to know the error with code and subcode, we can conclude that
+  // this error will have a particular status, statTags
+  code1: {
+    subCode1: {
+      status: ,
+      // Only tags that are unique to this error needs to be provided
+      statTags: 
+    }
+  }
+}
+ */
+const errorDetailsMap = {
+  100: {
+    // This error talks about event being sent after seven days or so
+    // Kind of a "badEvent" case?
+    2804003: {
+      status: 400,
+      statTags: {
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.ABORTABLE
+      }
+    }
+  },
+  1: {
+    // This error talks about event being sent after seven days or so
+    // Kind of a "badEvent" case?
+    99: {
+      status: 400,
+      statTags: {
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+        meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.META.ABORTABLE
+      }
+    }
+  }
+};
+
+const getErrorDetailsFromErrorMap = error => {
+  const { code, error_subcode: subCode } = error;
+  if (!isEmpty(errorDetailsMap[code]) && subCode) {
+    return errorDetailsMap[code][subCode];
+  }
+  // This is done to fix the eslint error, if no return is defined
+  return undefined;
```

Comment:
To avoid dummy return at the end to avoid the eslint error.
```suggestion
  const { code, error_subcode: subCode } = error;
  let errDetails;
  if (!isEmpty(errorDetailsMap[code]) && subCode) {
    errDetails = errorDetailsMap[code][subCode];
  }
  return errDetails;
```

Reply:
Changed. Can you let me know if it is ok ?